src/linux/sensor_ds18b20.c

```diff
@@ -247,8 +260,15 @@ ds18_send_and_request(struct ds18_s *d, uint32_t next_begin_time, uint8_t oid)
         if (request_time.tv_sec - d->request_time.tv_sec > W1_READ_TIMEOUT_SEC)
         {
             pthread_mutex_unlock(&d->lock);
-            try_shutdown("DS18B20 sensor didn't respond in time");
-            return;
+            //no response yet
+            d->error_count++;
```

Comment:
It might make sense to leave this part out and always fail on read timeouts. Errors in the read thread set a state of W1_ERROR which in turn sets the state back to W1_IDLE then W1_READ_REQUESTED (or as I suggested above skip W1_IDLE), and this also resets the read timeout time. So to get to this point we would have been waiting for a response for 5 seconds without encountering an error in the read thread. It seems likely to be a different kind of error than the transient ones that the error counter is addressing.

If keeping this behavior, the pthread_mutex_unlock shouldn't come before the read and writes to d->error_count.